drivers/video/video_esp32_dvp.c

```diff
@@ -57,8 +72,29 @@ struct video_esp32_data {
 	struct k_fifo fifo_in;
 	struct k_fifo fifo_out;
 	struct dma_block_config dma_blocks[CONFIG_DMA_ESP32_MAX_DESCRIPTOR_NUM];
+#ifdef CONFIG_POLL
+	struct k_poll_signal* signal_out;
+#endif
+#ifdef CONFIG_VIDEO_ESP32_VSYNC_INTERRUPT
+	struct k_poll_signal* signal_all;
```

Comment:
Should this not be inside `#ifdef CONFIG_POLL` along iwth `signal_out`?

Reply:
`CONFIG_VIDEO_ESP32_VSYNC_INTERRUPT` depends on `CONFIG_POLL`. 
https://github.com/zephyrproject-rtos/zephyr/blob/0e3e7e30dab280532542a4f1cdbd41cad576e169/drivers/video/Kconfig.esp32_dvp#L14

Reply:
Then the "nesting" is already ensured, and the code is a bit more legible without the nested `#if` macros.
Thanks for the explanation!